src/chunk_append/chunk_append.c

```diff
@@ -281,3 +291,63 @@ contain_param_exec_walker(Node *node, void *context)
 
 	return expression_tree_walker(node, contain_param_exec_walker, context);
 }
+
+typedef struct
+{
+	Var *sort_var;
+	Var *ht_var;
+	Index sort_relid;
+	Index ht_relid;
+	Oid eq_opr;
+} find_equality_join_var_context;
+
+static bool
+find_equality_join_var_walker(Node *node, find_equality_join_var_context *context)
+{
+	if (node == NULL)
+		return false;
+
+	if (IsA(node, OpExpr))
```

Comment:
Would this work if we have a subquery pulled up? Wondering if the jointree walker will handle it correctly.